### PR TITLE
Feature/view aatf details

### PR DESCRIPTION
**Pull request by willdot**
PBI: 70253

Not sure what kind of unit tests to do for the controller?

### CODE REVIEW

File: src/EA.Weee.RequestHandlers/Admin/GetAatfs/GetAatfInfoByAatfIdRequestHandler.cs [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+            this.aatfDataAccess = aatfDataAccess;   
+        }
+
+        public async Task<AatfData> HandleAsync(GetAatfById message)
+        {
+            authorization.EnsureCanAccessInternalArea();
+
+            var aatf = await aatfDataAccess.GetAatfById(message.AatfId);
+
+            if (aatf == null)
```
Comment (JB12298) [inline src/EA.Weee.RequestHandlers/Admin/GetAatfs/GetAatfInfoByAatfIdRequestHandler.cs]: Could you add a test for the null check?

File: src/EA.Weee.Web/Areas/Admin/Controllers/AatfController.cs [hunk 2]
```diff
@@ -32,6 +33,36 @@ public AatfController(Func<IWeeeClient> apiClient, IWeeeCache cache, BreadcrumbS
         }
 
         [HttpGet]
+        public async Task<ActionResult> Details(Guid id)
+        {
```
Comment (JB12298) [inline src/EA.Weee.Web/Areas/Admin/Controllers/AatfController.cs]: Missing controller tests

File: src/EA.Weee.RequestHandlers/Admin/GetAatfs/GetAatfInfoByAatfIdRequestHandler.cs (hunk 1 above)
Comment (willdot) [reply src/EA.Weee.RequestHandlers/Admin/GetAatfs/GetAatfInfoByAatfIdRequestHandler.cs]: Done

File: src/EA.Weee.Web/Areas/Admin/Controllers/AatfController.cs (hunk 2 above)
Comment (willdot) [reply src/EA.Weee.Web/Areas/Admin/Controllers/AatfController.cs]: Added some tests as well as mapping the view model with tests

File: src/EA.Weee.RequestHandlers.Tests.Unit/Admin/Aatf/GetAatfInfoByAatfIdRequestHandlerTests.cs [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+            Aatf returnData = null;
+
+            A.CallTo(() => dataAccess.GetAatfById(A.Dummy<Guid>())).Returns(returnData);
+
+            Func<Task> action = async () => await handler.HandleAsync(A.Dummy<GetAatfById>());
+
+            await Assert.ThrowsAsync<ArgumentException>(action);
+        }
+    }
+}
```
Comment (JohnPadgettCivica) [inline src/EA.Weee.RequestHandlers.Tests.Unit/Admin/Aatf/GetAatfInfoByAatfIdRequestHandlerTests.cs]: Add test to check that the value returned from the mapper is the value returned from the HandleAsync method

File: src/EA.Weee.Core/AatfReturn/UKCompetentAuthority.cs [hunk 4]
```diff
@@ -0,0 +1,6 @@
+﻿namespace EA.Weee.Core.AatfReturn
+{
+    public class UKCompetentAuthority
```
[the remark this answers is not in the document]
Comment (willdot) [reply src/EA.Weee.Core/AatfReturn/UKCompetentAuthority.cs]: That class is empty, so not sure what it's there for. Have removed.

**Comment by JohnPadgettCivica** [comment]
Seems just need to merge from develop as some conflicting changes
